### ISSUE TITLE
`vue/define-macros-order` false positive with using `export`

### ISSUE THREAD

**Issue by Ky6uk** [vuejs/eslint-plugin-vue#1981]
`vue/define-macros-order` produces an error every time I'm trying to add an `export` to `Props`. Exporting props [is one of the ways](https://vuejs.org/api/sfc-script-setup.html#default-props-values-when-using-type-declaration) how to use TypeScript with Vue.

It looks like the rule doesn't handle type definitions as type definitions in case we're trying to export them. 🤔 

**Tell us about your environment**

- **ESLint version:** 8.23.1
- **eslint-plugin-vue version:** 9.5.1
- **Node version:** 18.8.0
- **Operating System:** Mac OS

**Please show your full configuration:**

```json5
{
    "parserOptions": {
        "parser": "@typescript-eslint/parser",
        "extraFileExtensions": [".vue"]
    },

    "extends": ["plugin:vue/base"],

    "rules": {
        "vue/define-macros-order": "error"
    }
}
```

**What did you do?**	

```vue
<script setup lang="ts">
  export interface Props {}

  defineProps<Props>();
</script>
```

**What did you expect to happen?**

No error was produced.

**What actually happened?**

```
/private/tmp/eslint-vue-define-macros-order-bug/App.vue
  4:3  error  defineProps should be the first statement in `<script setup>` (after any potential import statements or type definitions)  vue/define-macros-order

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Repository to reproduce this issue**

https://github.com/Ky6uk/eslint-vue-define-macros-order-bug

**Comment by FloEdelmann** [comment]
Indeed, it looks like type definitions that are exported will erroneously be reported as errors. A pull request to fix this is welcome :slightly_smiling_face: 

Please also add a test case (and maybe a fix) for `export interface Foo {}`.

**Comment by FloEdelmann** [comment]
@Ky6uk Note that in your case, you probably don't need to export the type definition. Just declaring it should probably also work fine, and should not cause this issue.

**Comment by Ky6uk** [comment]
@FloEdelmann yep, it's not a blocker for sure. Unfortunately, there's another issue [related to Volar](https://github.com/johnsoncodehk/volar/issues/1232) in case the export is omitted. ¯\_(ツ)_/¯

**Comment by Ky6uk** [comment]
@FloEdelmann just to note that I've tested it and it looks like this bug has been fixed in the `v9.8.0`. At least I can't reproduce it anymore since `v9.8.0`. 🤔
